src/cli/argument-parser.js

```diff
@@ -0,0 +1,211 @@
+        var remoteMatch = browser.match(REMOTE_ALIAS_RE);
+
+        if (remoteMatch) {
+            this.remoteCount += parseInt(remoteMatch[1], 10) || 1;
+            return false;
+        }
+
+        return true;
+    }
+
+    _parseFilteringOptions () {
+        this.opts.testGrep    = CliArgumentParser._optionValueToRegExp('--test-grep', this.opts.testGrep);
+        this.opts.fixtureGrep = CliArgumentParser._optionValueToRegExp('--fixture-grep', this.opts.fixtureGrep);
+
+        this.filter = (testName, fixtureName) => {
+
+            if (this.opts.test && testName !== this.opts.test)
+                return false;
+
+            if (this.opts.testGrep && !this.opts.testGrep.test(testName))
+                return false;
+
+            if (this.opts.fixture && fixtureName !== this.opts.fixture)
+                return false;
+
+            if (this.opts.fixtureGrep && !this.opts.fixtureGrep.test(fixtureName))
+                return false;
+
+            return true;
+        };
+    }
+
+    async _parsePorts () {
+        if (this.opts.ports) {
+            this.opts.ports = this.opts.ports
+                .split(',')
+                .map(CliArgumentParser._parsePortNumber);
+
+            if (this.opts.ports.length < 2)
+                throw new Error(getText(MESSAGE.portsOptionRequiresTwoNumbers));
+        }
+    }
+
+    async _parseBrowserList () {
+        var browsersArg   = this.program.args[0] || '';
+        var installations = await getBrowserInstallations();
+        var allAliases    = Object.keys(installations);
+
+        this.browsers = browsersArg
+            .split(',')
+            .filter(browser => browser && this._filterAndCountRemotes(browser))
+            .reduce((browserList, browser) => CliArgumentParser._replaceAllBrowsersAlias(browserList, browser, allAliases), []);
+    }
+
+    async _parseFileList () {
+        var fileList = this.program.args.slice(1);
+
+        this.src = await globby(fileList, {
+            cwd:    this.cwd,
+            silent: true,
+            nodir:  true
+        });
+
+        this.src = this.src.map(file => resolve(this.cwd, file));
+    }
+
+    async _parseScreenshotsPath () {
+        if (this.opts.screenshots) {
+            this.opts.screenshots = resolve(this.cwd, this.opts.screenshots);
+
+            await ensureDir(this.opts.screenshots);
+        }
+    }
+
+    async _parseReportPath () {
+        if (this.opts.reportPath) {
+            this.opts.reportPath = resolve(this.cwd, this.opts.reportPath);
+
+            var dir = dirname(this.opts.reportPath);
+
+            await ensureDir(dir);
+
+            this.reportOutStream = createWriteStream(this.opts.reportPath);
+        }
+    }
+
+    async parse (argv) {
+        this.program.parse(argv);
+
+        this.opts = this.program.opts();
+
+        // NOTE: '-list-browsers' option just lists browsers and immediately exits app.
+        // Therefore, we don't need process other args
```

Comment:
NOTE: the '-list-browsers' option only lists browsers and immediately exits the app.
Therefore, we don't need to process other arguments.